internal/go/skycfg/proto_message.go

```diff
@@ -64,6 +65,58 @@ func (msg *skyProtoMessage) Freeze() {
 	}
 }
 
+func (msg *skyProtoMessage) CompareSameType(op syntax.Token, y starlark.Value, depth int) (bool, error) {
+	other, ok := y.(*skyProtoMessage)
+	if !ok {
+		return false, nil
+	}
+
+	switch op {
+	case syntax.EQL:
+		return msg.equals(other, depth)
+	case syntax.NEQ:
+		eql, err := msg.equals(other, depth)
+		if err != nil {
+			return eql, err
+		}
+
+		return !eql, err
+	default:
+		return false, fmt.Errorf("Only == and != operations are supported on protobufs, found %s", op.String())
+	}
+}
+
+func (msg *skyProtoMessage) equals(other *skyProtoMessage, depth int) (bool, error) {
+	msgAttrNames := msg.AttrNames()
```

Comment:
Lets use [`proto.Equal()`](https://godoc.org/github.com/golang/protobuf/proto#Equal) to handle the Protobuf equality.